libs/execution_context/ebpf_program.c

```diff
@@ -138,13 +148,27 @@ _ebpf_program_program_info_provider_changed(
             // A program info provider cannot modify helper function count upon reload.
             goto Exit;
         }
+        if ((program->global_helper_function_count > 0) &&
+            (helper_function_addresses->helper_function_count != program->provider_helper_function_count)) {
+
+            EBPF_LOG_MESSAGE_GUID(
+                EBPF_TRACELOG_LEVEL_ERROR,
+                EBPF_TRACELOG_KEYWORD_PROGRAM,
+                "A program info provider cannot modify global helper function count upon reload",
```

Comment:
Ok, but why can't it increase it?  (Same question on 141)
That is why `!=` instead of just `<`?

Reply:
It seems this check is due to the way trampoline table works right now. Once trampoline table is initialized for `n` entries, it expects exactly `n` entries in the next update also. We can change this behavior if we think adding new helper functions on the fly should be allowed.